lib/common.js

```diff
@@ -145,20 +145,28 @@ class ExpiringMap extends Map {
 
 const utf16singleUnit = 1 << 16;
 
+// https://tools.ietf.org/html/rfc3629#section-3
+const utf8bytesLastCodePoints = {
+  1: 0x007F,
+  2: 0x07FF,
+  3: 0xFFFF,
+};
+
 const utf8split = (str, maxByteCount) => {
   let utf8length = 0;
   for (let i = 0; i < str.length;) {
     const codePoint = str.codePointAt(i);
-    let utf8codePointSize;
-    if (codePoint < 0x80) {
-      utf8codePointSize = 1;
-    } else if (codePoint < 0x800) {
-      utf8codePointSize = 2;
-    } else if (codePoint < 0x10000) {
-      utf8codePointSize = 3;
-    } else {
-      utf8codePointSize = 4;
+
+    // Calculate code point size in bytes when encoded with UTF-8:
+    let utf8codePointSize = 4;
+    for (let byteCount = 1; byteCount <= 3; byteCount++) {
```

Comment:
Though no need to change it as the way it is done below is much more cryptic 🙃.
```js
let byteCount = 0;
while (++byteCount < 4 && codePoint > utf8bytesLastCodePoints[byteCount]);
const utf8codePointSize = byteCount;
```
Good ol' C =) We can even include `4` in the utf8bytesLastCodePoints and go even further.

Reply:
What's the point of including `4` in the `utf8bytesLastCodePoints`?
You can't get the `codePoint` higher than `0x10FFFF` which is the maximal Unicode code point that can be represented by 4 UTF-8 code units.

Reply:
Of I forgot it's `undefined` check. Then just
```js
let byteCount = 0;
while (codePoint > utf8bytesLastCodePoints[++byteCount]);
const utf8codePointSize = byteCount;
```
:upside_down_face: